core-java-modules/core-java-io-5/src/test/java/com/baeldung/outputstreamtobytearray/OutputStreamtoByteArrayUnitTest.java

```diff
@@ -0,0 +1,70 @@
+package com.baeldung.outputstreamtobytearray;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+
+public class OutputStreamtoByteArrayUnitTest {
+
+    @Test
+    public void givenByteArrayOutputStream_whenUsingToByteArray_thenReturnByteArray() throws IOException {
+        String data = "Welcome to Baeldung!";
+
+        try (ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream()) {
+            byteArrayOutputStream.write(data.getBytes());
+
+            byte[] byteArray = byteArrayOutputStream.toByteArray();
+            String result = new String(byteArray, StandardCharsets.UTF_8);
+
+            Assertions.assertEquals(data, result);
+        }
+    }
+
+    @Test
+    public void givenFileOutputStream_whenUsingIOUtilsToReadTheFile_thenReturnByteArray() throws IOException {
+        String data = "Welcome to Baeldung!";
+        String filePath = "file.txt";
+
+        try (FileOutputStream outputStream = new FileOutputStream(filePath)) {
+            outputStream.write(data.getBytes());
+        }
+
+        byte[] byteArray = FileUtils.readFileToByteArray(new File(filePath));
+        String result = new String(byteArray);
+
+        Assertions.assertEquals(data, result);
+    }
+
+    @Test
+    public void givenDrainableOutputStream_whenDraining_thenReturnByteArray() {
+        DrainableOutputStream drainableOutputStream = new DrainableOutputStream();
+        String data = "Welcome to Baeldung!";
+
+        try {
+            drainableOutputStream.write(data.getBytes());
+        } catch (IOException e) {
+            e.printStackTrace();

```

Comment:
Do not catch exceptions like this, they are not being tested and so you should simply add `throws IOException` to the test method signature